bin/clone

```diff
@@ -38,29 +32,46 @@ def main():
     parser.add_argument(
         "--css-validate", help="set whether css validation is required", type=str_to_bool, default=None
     )
+    parser.add_argument("--path", help="path to save the page to be cloned", required=False, default="/opt/")
+
     args = parser.parse_args()
+    default_path = os.path.join(args.path, 'snare')
+
     if args.log_path:
-        log_file = args.log_path + "clone.log"
+        log_file = os.path.join(args.log_path, "clone.log")
     else:
-        log_file = "/opt/snare/clone.log"
+        log_file = os.path.join(default_path, 'clone.log')
+
+    if(not os.path.realpath(default_path).startswith(os.path.expanduser('~')) or
```

Comment:
to test permissions, you can also easily try to write something into the base path and catch the exception.
A user might have permissions not only to his own folder